src/garage/torch/algos/td3.py

```diff
@@ -0,0 +1,429 @@
+        policy (garage.torch.policies.Policy): Policy (actor network).
+        qf1 (garage.torch.q_functions.QFunction): Q function (critic network).
+        qf2 (garage.torch.q_functions.QFunction): Q function (critic network).
+        replay_buffer (ReplayBuffer): Replay buffer.
+        replay_buffer_size (int): Size of the replay buffer
+        exploration_policy (garage.np.exploration_policies.ExplorationPolicy):
+                Exploration strategy.
+        uniform_random_policy
+                (garage.np.exploration_policies.ExplorationPolicy):
+                Uniform random exploration strategy.
+        target_update_tau (float): Interpolation parameter for doing the
+            soft target update.
+        discount (float): Discount factor (gamma) for the cumulative return.
+        reward_scaling (float): Reward scaling.
+        update_actor_interval (int): Policy (Actor network) update interval.
+        max_action (float): Maximum action magnitude.
+        buffer_batch_size (int): Size of replay buffer.
+        min_buffer_size (int): The minimum buffer size for replay buffer.
+        policy_noise (float): Policy (actor) noise.
+        policy_noise_clip (float): Noise clip.
+        exploration_noise (float): Exploration noise.
+        clip_return (float): Clip return to be in [-clip_return,
+            clip_return].
+        policy_lr (float): Learning rate for training policy network.
+        qf_lr (float): Learning rate for training Q network.
+        policy_optimizer (Union[type, tuple[type, dict]]): Type of optimizer
+            for training policy network. This can be an optimizer type such as
+            `torch.optim.Adam` or a tuple of type and dictionary, where
+            dictionary contains arguments to initialize the optimizer
+            e.g. `(torch.optim.Adam, {'lr' : 1e-3})`.
+        qf_optimizer (Union[type, tuple[type, dict]]): Type of optimizer
+            for training Q-value network. This can be an optimizer type such
+            as `torch.optim.Adam` or a tuple of type and dictionary, where
+            dictionary contains arguments to initialize the optimizer
+            e.g. `(torch.optim.Adam, {'lr' : 1e-3})`.
+        steps_per_epoch (int): Number of train_once calls per epoch.
+        grad_steps_per_env_step (int): Number of gradient steps taken per
+            environment step sampled.
+        num_evaluation_episodes (int): The number of evaluation
+            trajectories used for computing eval stats at the end of every
+            epoch.
+        start_steps (int): The number of steps for warming up before
+             selecting actions according to policy.
+        update_after (int): The number of steps to perform before policy
+            is updated.
+
+    """
+
+    def __init__(
+        self,
+        env_spec,
+        policy,
+        qf1,
+        qf2,
+        replay_buffer,
+        grad_steps_per_env_step,
+        exploration_policy=None,
```

Comment:
Does this algorithm work at all if `exploration_policy` isn't passed? If it's actually required, don't give it a default value.